src/wasm/wasm-type.cpp

```diff
@@ -27,14 +27,20 @@
 template<> class std::hash<std::vector<wasm::Type>> {
 public:
   size_t operator()(const std::vector<wasm::Type>& types) const {
```

Comment:
`typeLists` needs to store vectors of types because those are the interned vectors. If `typeLists` did not store the vectors, there would not be a way to query the value types that make up a `Type`. `indices` similarly needs to be keyed on vectors of types because we need to be able to look up the canonical type (i.e. index) for any given vector of value types. That's why we need to be able to hash vectors of types. That hash is not exposed in the public interface.